docs/source/user-guide/ir/types.rst

```diff
@@ -126,23 +127,25 @@ When Typed Pointers are enabled, the pointer type is represented using:
 
         The type pointed to.
 
-Opaque pointers can be enabled by setting the environment variable:
+Opaque Pointers can be enabled in the IR layer by setting the environment
+variable:
 
 .. code:: bash
 
-   LLVMLITE_ENABLE_OPAQUE_POINTERS=1
+   LLVMLITE_ENABLE_IR_LAYER_TYPED_POINTERS=0
 
-or by setting the ``opaque_pointers_enabled`` attribute after importing
+or by setting the ``ir_layer_typed_pointers_enabled`` attribute after importing
 llvmlite, but prior to using any of its functionality. For example:
 
 .. code:: python
 
    import llvmlite
-   llvmlite.opaque_pointers_enabled = True
+   llvmlite.ir_layer_typed_pointers_enabled = False
 
    # ... continue using llvmlite ...
 
-When Opaque Pointers are enabled, the pointer type is represented using:
+When Opaque Pointers are enabled (by disabling Typed Pointers), the pointer
+type is represented using:
```

Comment:
Whilst this is _technically_ true, typed pointers won't be represented as "true" opaque pointers in the sense that they'll still be represented as `_TypedPointerType` regardless of `ir_layer_typed_pointers_enabled`.

This shouldn't matter for users of PointerType as everything should still work transparently (that's the whole point of having the shim). For all intended purposes, all pointer types are always `PointerType`.

Reply:
Actually reading your second comment below, I'm rethinking this. Will push an update that I hope is a bit more representative and helpful.

Reply:
I've tried to decouple the wording from whether the lazy mode is enabled or not, in the hope it simplifies things - let me know what you think.

Reply:
Thanks very much, I think the way you've broken it down is spot on.